### DISCUSSION TITLE
🔎 Code Search Q&A

### DISCUSSION THREAD

**Discussion by colinwm** [community/events#59] [Cloud]
Any questions about our new code search and code view beta? https://github.com/features/code-search

Ask them here and we'll answer them!

**Comment by JackEAllen** [comment]
Can you filter by file type? Trying to filter by language hasn't worked previously for me when working with niche frameworks such as [EPICS](https://epics-controls.org/) which uses `.db` and `.proto` files built on C and It is a popular framework for large scale scientific facilities for controls development.

**Reply by colinwm** [reply]
Yes! You can filter by file extension like this:

```
path:*.proto
```

Or you can use language filters:

```
lang:protobuf
```

And you can even use boolean expressions to find several different extensions/languages/filetypes in a single query, like:

```
(lang:protobuf OR path:*.db OR path:*.txt) search_term
```

**Comment by osppro** [comment]
If you are looking to filter files by their type, such as filtering by .db and .proto files used in EPICS which are built on C language, you may need to use a file explorer or a search tool that supports filtering by file extensions. EPICS is indeed a widely used framework in large-scale scientific facilities for controls development. If you encounter issues filtering by language, focusing on the file types directly could be a more effective approach.

**Reply by JackEAllen** [reply]
My query has already been resolved by @colinwm - See quoted reply below:
> Yes! You can filter by file extension like this:
> 
> ```
> path:*.proto
> ```
> 
> Or you can use language filters:
> 
> ```
> lang:protobuf
> ```
> 
> And you can even use boolean expressions to find several different extensions/languages/filetypes in a single query, like:
> 
> ```
> (lang:protobuf OR path:*.db OR path:*.txt) search_term
> ```

**Comment by xTalhaAhmedCho** [comment]
![chrome_eTjRY3DcEQ](https://github.com/user-attachments/assets/247865c8-f382-48c5-a43a-de6bd50d3803)

`Dear GitHub Support Team,`

> I am writing to request the re-enablement of GitHub Actions for my account, **@TalhaAhmedCho**. I have noticed that GitHub Actions is currently disabled for my account. 

I understand that my GitHub Actions may have been disabled due to `misunderstanding of limits`. If my actions inadvertently violated any GitHub terms or policies, I sincerely apologize. It was not my intention to misuse the platform. I am committed to using GitHub Actions responsibly and in accordance with all guidelines in the future.

> I value GitHub's services greatly, especially GitHub Actions, which is an **_integral part of my workflow_** for personal project. I rely on GitHub Actions for `learning new technologies. `

**_I kindly request you to review my account and re-enable GitHub Actions. I am willing to cooperate fully and take any necessary steps to ensure I comply with GitHub's policies and guidelines in the future._**

`My GitHub username is:`  **@TalhaAhmedCho**

Thank you for your **_time and consideration_**. I look forward to your positive response and the re-enablement of GitHub Actions.

Sincerely,

Talha Ahmed Chowdhury
[Linkedin](https://www.linkedin.com/in/TalhaAhmedCho/) | [Facebook](https://www.facebook.com/@TalhaAhmedCho) | [YouTube](https://www.youtube.com/@TalhaAhmedCho) | [Twitter](https://x.com/_TalhaAhmedCho)
